### PR TITLE
[Model] - Create Installment Model

### PR DESCRIPTION
**Pull request by vinibr3**
Created Installment model and his tests.

### CODE REVIEW

File: spec/models/installment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  it { should validate_numericality_of(:value).is_greater_than_or_equal_to(0) }
+  it { should validate_presence_of(:value) }
+
+  it { should validate_presence_of(:due_date) }
+
+  it { should validate_presence_of(:payment_date) }
+
+  it { should validate_numericality_of(:total_value).is_greater_than_or_equal_to(0) }
+  it { should validate_presence_of(:total_value) }
+  context do
```
Comment (RodrigoNM) [inline spec/models/installment_spec.rb]: You should write a description to your context, as the rspec best practices suggest, when describing a context, start its description with "when" or "with".

File: spec/models/installment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+RSpec.describe Installment, type: :model do
+  it { should validate_numericality_of(:off).is_greater_than_or_equal_to(0) }
```
Comment (RodrigoNM) [inline spec/models/installment_spec.rb]: Use expect instead of should.

File: spec/factories.rb [hunk 3]
```diff
@@ -1,4 +1,14 @@
 FactoryBot.define do
+  factory :installment do
+    digit = Faker::Number.digit
+    value = Faker::Number.decimal(digit.to_i, 2)
+
+    value { value }
```
Comment (RodrigoNM) [inline spec/factories.rb]: Why you didn't use faker here?

**Review by RodrigoNM — DISMISSED** [review]

**Review by RodrigoNM — DISMISSED** [review]

**Review by Heirian — DISMISSED** [review]

**Review by RodrigoNM — DISMISSED** [review]

File: app/models/installment.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  validates :value, numericality: { greater_than_or_equal_to: 0 }, presence: true
+  validates :total_value, numericality: { greater_than_or_equal_to: 0 }, presence: true
+  validates :payment_date, presence: true
+  validates :due_date, presence: true
+
+  before_validation :calculate_total_value
+
+  private
+
+    def calculate_total_value
```
Comment (Heirian) [inline app/models/installment.rb:12]: [Indent the public, protected, and private methods as much the method definitions they apply to.](https://github.com/airbnb/ruby#access-modifiers)

File: spec/factories/installment.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :installment do
+    digit = Faker::Number.digit
+    value = Faker::Number.decimal(digit.to_i, 2)
+    value { value }
```
Comment (Heirian) [inline spec/factories/installment.rb]: Why?

**Review by Heirian — CHANGES_REQUESTED** [review]

File: spec/factories/installment.rb (hunk 5 above)
Comment (vinibr3) [reply spec/factories/installment.rb]: decimal method create an number with 'digit' length randomic to left dot and length 2 after dot 
An better aproach is Faker::Number.positive.round(2), means @RodrigoNM

File: app/models/installment.rb (hunk 4 above)
Comment (vinibr3) [reply app/models/installment.rb:12]: After check, it broke in rubocop and CI.

Comment (Heirian) [reply app/models/installment.rb:12]: I saw. You'are using `rubocop-rails_config` gem to use the `rails guidelines`. Isn't a big deal.
We use the standard rubocop guidelines here though.

**Review by Heirian — APPROVED** [review]

**Review by kaiomagalhaes — APPROVED** [review]

**Review by RodrigoNM — APPROVED** [review]

**Pull request merged**